### PR TITLE
docs: Fix and improve networking sections

### PR DESCRIPTION
**Pull request by okurz**
The two sections "TAP Based Network" and "VDE Based Network" were below
"QEMU User Networking" which is wrong. Over time the section "Multi
Machine Tests Setup" was added which is specific to the use of
openvswitch equivalent to "TAP Based Network" so we should combine both
sections. And as "VDE Based Network" is less used I move it to the
bottom.

Related progress issue: https://progress.opensuse.org/issues/33697

### CODE REVIEW

**Comment by okurz** [comment]
I added a commit "docs: Extend Networking chapter to cover non-qemu in general"

File: docs/Networking.asciidoc [hunk 1]
```diff
@@ -5,16 +5,20 @@
...
+For tests using the QEMU backend the networking type used is controlled by the
+`NICTYPE` variable. If unset or empty `NICTYPE` defaults to `user`, i.e.
+<<QEMU User Networking>> which requires no further configuration.
 
 For more advanced setups or tests that require multiple jobs to be in the same
 networking the <<TAP based network,TAP>> or <<VDE Based Network,VDE>> based
 modes can be used.
 
+Other backends can be treated just the same as bare-metal setups. Tests can be
+triggered in parallel same for QEMU based ones and synchronization primitives
```
Comment (Martchus) [inline docs/Networking.asciidoc:17]: Is some connecting word between "parallel" and "same" missing?

File: docs/Networking.asciidoc [hunk 2]
```diff
@@ -5,16 +5,20 @@
...
+can be used. As the network can not be fully virtual of course then the network
```
Comment (Martchus) [inline docs/Networking.asciidoc:18]: The "then" doesn't make it better.

File: docs/Networking.asciidoc (hunk 1 above)
Comment (okurz) [reply docs/Networking.asciidoc:17]: Actually the word "as" is missing after "same" :)

File: docs/Networking.asciidoc (hunk 2 above)
Comment (okurz) [reply docs/Networking.asciidoc]: Rephrased

**Pull request merged**